docs/conf.py

```diff
@@ -55,7 +55,7 @@
 extensions = [
     'sphinx.ext.autodoc',
     'sphinx.ext.todo',
-    'sphinx.ext.viewcode',
+#    'sphinx.ext.viewcode',
```

Comment:
@guolinke So, did commenting this help to pass docs test?

Reply:
yeah, it can pass now

Reply:
It means that the reason was in these auto-generated links by Sphinx. To be honest, I don't know what is better: completely disable them for tests or use old LinkChecker...

![image](https://user-images.githubusercontent.com/25141164/42411507-7961072a-8205-11e8-8353-907bf3a8afbe.png)

Reply:
What will happen if disable them ?

Reply:
Something like this:

![image](https://user-images.githubusercontent.com/25141164/42412114-e7faade0-820e-11e8-814e-4103f805835f.png)

No possibility to use links with anchors to concrete method and users cannot quickly check the code of corresponding method (`[Source]` button).

I think it's unacceptable and can be used only for tests at CI side, not at ReadTheDocs.

Reply:
Maybe better to pin the old version of LinkChecker for now?..

Reply:
okay, I don't how to pin it. can you help with that ?

Reply:
Sure! Will do this.

Reply:
To be honest, it's very painful to do with without possibility to cancel build and need to wait all jobs are finished (macOS is running more that 1 hour due to brew problems).

I suppose that we don't need to run `doc-test` at VSTS for now. Let Travis to perform this.